doc/tutorial/security.dox

```diff
@@ -0,0 +1,247 @@
+/**
+ * @page security Securing the communications
+ * @tableofcontents
+ *
+ * @section enabling-network-layer-security Enabling network layer security
+ *
+ * Network layer security can be used to secure unicast communications
+ * between two nodes (a single hop).  This includes the subscription,
+ * acknowledgement, and subscription acknowledgement messages.  When
+ * the two nodes are @link link linked @endlink, it also includes
+ * publications.  Multicast publications are not secured.  One of the
+ * end-to-end mechanisms described below must be used to secure
+ * multicast publications.
+ *
+ * The first step to enabling network layer security is to build with
+ * a transport that supports it.
+ *
+ * @code
+ * $ scons transport=dtls
+ * @endcode
+ *
+ * The DTLS transport supports two mechanisms for securing the
+ * connection: pre-shared keys, and certificates.
+ *
+ * @subsection dtls-with-pre-shared-keys DTLS with pre-shared keys
+ *
+ * @snippet tutorial.c Pre-shared key
+ * To use pre-shared keys (PSKs), we'll need to agree on a PSK and its
+ * identifier.
+ *
+ * @snippet tutorial.c Creating a secure node with a network PSK
+ * All of the security mechanisms require that the node be created
+ * with a key store.  The DTLS PSK mechanism requires that a @ref
+ * DPS_KeyAndIdHandler and @ref DPS_KeyHandler be provided.
+ *
+ * @snippet tutorial.c PSK and ID handler
+ * The key and identifier handler simply uses DPS_SetKeyAndId() to
+ * return the PSK and identifier we agreed on earlier.
+ *
+ * @snippet tutorial.c PSK handler
+ * The key handler is used for PSKs as well as other keys, so it must
+ * compare the incoming @c keyId against the PSK identifier and either
+ * return the PSK using DPS_SetKey() or return @ref DPS_ERR_MISSING.
+ *
+ * @see DPS_CreateMemoryKeyStore(), DPS_SetNetworkKey()
+ *
+ * @subsection dtls-with-certificates DTLS with certificates
+ *
+ * @snippet tutorial.c Certificates
+ * To use certificates, we'll need a certificate for each node and the
+ * certificates of the authorities that issued them.  DPS supports
+ * Elliptic Curve Cryptography (ECC) certificates.  The certificates
+ * above are elided due to their size.
```

Comment:
I would used the word `omitted` in place of elided. I feel its a much more common word. I had to actually look up elided. Still it is accurate so I will leave this a preference comment.

Reply:
No problem, I can replace with omitted.